Modules/Messages/src/Messages.Application/Messages/DTOs/MessageDTO.cs

```diff
@@ -1,26 +1,48 @@
-using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.Mapping;
 using Backbone.DevelopmentKit.Identity.ValueObjects;
 using Backbone.Modules.Messages.Domain.Entities;
 using Backbone.Modules.Messages.Domain.Ids;
 
 namespace Backbone.Modules.Messages.Application.Messages.DTOs;
 
-public class MessageDTO : IMapTo<Message>
+public class MessageDTO
 {
-    public required MessageId Id { get; set; }
+    public MessageDTO(Message message, IdentityAddress activeIdentity, string didDomainName)
+    {
+        Id = message.Id;
+        CreatedAt = message.CreatedAt;
+        CreatedBy = message.CreatedBy;
+        CreatedByDevice = message.CreatedByDevice;
+        Body = message.Body;
+        Attachments = message.Attachments.Select(a => new AttachmentDTO(a)).ToList();
+        Recipients = MapRecipients(message, activeIdentity, didDomainName);
+    }
+
+    public MessageId Id { get; set; }
+    public DateTime CreatedAt { get; set; }
+    public IdentityAddress CreatedBy { get; set; }
+    public DeviceId CreatedByDevice { get; set; }
+    public byte[] Body { get; set; }
+    public List<AttachmentDTO> Attachments { get; set; }
+    public List<RecipientInformationDTO> Recipients { get; set; }
 
-    public required DateTime CreatedAt { get; set; }
-    public required IdentityAddress CreatedBy { get; set; }
-    public required DeviceId CreatedByDevice { get; set; }
+    private static List<RecipientInformationDTO> MapRecipients(Message message, IdentityAddress activeIdentity, string didDomainName)
```

Comment:
I find it strange having this much business logic in the DTO, should we not move it elsewhere?

Reply:
Yeah, I was struggling with that, too... I think that the logic has to be executed by the MessageDTO. Meaning that we should not rely on the caller (aka the Handler) to call the logic. 
I could extract this into a separate class. But would this help? 

I am up for discussion.

(keep in mind that the logic is at least tested by the integration tests)

Reply:
I just added some tests for the DTO.